controllers/vm_migration_controller.go

```diff
@@ -0,0 +1,193 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"strings"
+
+	networkv1alpha1 "github.com/NCCloud/tabby-cni/api/v1alpha1"
+	"github.com/j-keck/arping"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	virtv1 "kubevirt.io/api/core/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+// VirtualMachineReconciler makes sure the VM has a correct mac address on the switch
+type VirtualMachineReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+}
+
+//+kubebuilder:rbac:groups=kubevirt.io,resources=virtualmachineinstances,verbs=get;list;watch
+
+func (r *VirtualMachineReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	_ = log.FromContext(ctx)
+
+	log.Log.Info(fmt.Sprintf("VirtualMachine: Reconcile Virtual Machine %+v", req))
+
+	virtualMachineInstance := &virtv1.VirtualMachineInstance{}
+	err := r.Get(ctx, req.NamespacedName, virtualMachineInstance)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	virtualMachineNetworks := virtualMachineInstance.Spec.Networks
```

Comment:
There might be possible nil pointer dereference, not fully sure :)

Reply:
Yes, you are right:)
```
Networks []Network `json:"networks,omitempty"`
```
But from what I see, if you don't set `networks` during creation, it automatically populates the field with the pod network.
But let's imagine it doesn't. I think the default value will be an empty slice and iterating over it will just result in 0 iterations? :see_no_evil: